services/repository/adopt.go

```diff
@@ -143,8 +143,6 @@ func adoptRepository(ctx context.Context, repoPath string, u *user_model.User, r
 				return fmt.Errorf("setDefaultBranch: %v", err)
 			}
 		}
-
-		repo.DefaultBranch = strings.TrimPrefix(repo.DefaultBranch, git.BranchPrefix)
```

Comment:
I'm not sure I agree...

```go
// GetDefaultBranch gets default branch of repository.
func (repo *Repository) GetDefaultBranch() (string, error) {
	stdout, _, err := NewCommand(repo.Ctx, "symbolic-ref", "HEAD").RunStdString(&RunOpts{Dir: repo.Path})
	return stdout, err
}
```

```bash
git checkout main 
git symbolic-ref HEAD
```

Reply:
Updated https://github.com/go-gitea/gitea/pull/20369/files/1ee01e00e298af3db57c30fbcb3789dc9cf181cf..89e5e994ed90823f8bf57743cc16177143d28d99

Now `GetDefaultBranch` returns the branch name, not the ref name. This is the only place calling GetDefaultBranch.